rln-prover/prover_db_migration/src/m20251128_nullifiers.rs

```diff
@@ -0,0 +1,74 @@
+use sea_orm_migration::prelude::*;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        // Create nullifiers table - optimized for high-throughput duplicate detection
+        // Primary use case: check if (nullifier, epoch) exists (hot path - 500+ TPS)
+        // Secondary: cleanup old epochs
+        //
+        // Design decisions for performance:
+        // 1. Composite primary key (nullifier, epoch) - single index lookup
+        // 2. No auto-increment ID - saves write overhead
+        // 3. Nullifier as BYTEA(32) - compact storage, fast comparison
+        // 4. Epoch as BIGINT - supports timestamps or block numbers
+        manager
+            .create_table(
+                Table::create()
+                    .table(Nullifiers::Table)
+                    // Nullifier: 32 bytes from RLN proof (internal nullifier)
```

Comment:
Note that there is no constraint applied in Postgresql so maybe this is worth to be mentionned (mention that this should be check in code)?